test/utils.ts

```diff
@@ -0,0 +1,219 @@
+import { Block, Provider } from '@ethersproject/providers';
+import axios from 'axios';
+import { Capsule, CapsuleWithFrags, reencrypt, VerifiedCapsuleFrag, VerifiedKeyFrag } from 'umbral-pre';
+
+import { Alice, Bob } from '../src';
+import { StakingEscrowAgent } from '../src/agents/staking-escrow';
+import { GetUrsulasResponse, Porter, Ursula } from '../src/characters/porter';
+import { RetrievalResult } from '../src/kits/retrieval';
+import { TreasureMap } from '../src/policies/collections';
+import { HRAC } from '../src/policies/hrac';
+import { BlockchainPolicy } from '../src/policies/policy';
+import { ChecksumAddress, Configuration } from '../src/types';
+import { toBytes, zip } from '../src/utils';
+
+const mockConfig: Configuration = {
+  porterUri: 'https://_this_should_crash.com/',
+};
+
+export const mockBob = (): Bob => {
+  const bobKey = toBytes('fake-secret-key-32-bytes-bob-xxx');
+  return Bob.fromSecretKey(mockConfig, bobKey);
+};
+
+export const mockRemoteBob = (): Bob => {
+  const bob = mockBob();
+  return Bob.fromPublicKeys(
+    mockConfig,
+    bob.verifyingKey,
+    bob.decryptingKey,
+  );
+};
+
+export const mockAlice = () => {
+  const aliceKey = toBytes('fake-secret-key-32-bytes-alice-x');
+  const alice = Alice.fromSecretKey(mockConfig, aliceKey);
+  const provider = mockWeb3Provider();
+  alice.transactingPower.connect(provider as Provider);
+  return alice;
+};
+
+export const mockWeb3Provider = (
+  blockNumber?: number,
+  blockTimestamp?: number,
+): Partial<Provider> => {
+  const block = { timestamp: blockTimestamp ?? 1000 };
+  return {
+    getBlockNumber: () => Promise.resolve(blockNumber ?? 1000),
+    getBlock: () => Promise.resolve(block as Block),
+    _isProvider: true,
+    getNetwork: () => Promise.resolve({ name: 'mock', chainId: -1 }),
+  };
+};
+
+export const mockUrsulas = (): Ursula[] => {
+  return [
+    {
+      encryptingKey:
+        '025a335eca37edce8191d43c156e7bc6b451b21e5258759966bbfe0e6ce44543cb',
+      checksumAddress: '0x5cF1703A1c99A4b42Eb056535840e93118177232',
```

Comment:
If these are supposed to match what Porter returns exactly then `encryptingKey` -> `encrypting_key`, and `checksumAddress` -> `checksum_address`.